merlin-sdk/src/main/java/gov/nasa/jpl/aerie/merlin/protocol/types/UnconstructableArgumentException.java

```diff
@@ -0,0 +1,11 @@
+package gov.nasa.jpl.aerie.merlin.protocol.types;
+
+public final class UnconstructableArgumentException extends Exception {
```

Comment:
🔵 I suspect we could do away with this exception (since we're throwing it then immediately catching it), but I won't insist on that for this PR.

Reply:
Oh, yeah that try/catch is kind of annoying. It’s needed since the only way to get a failure string is to use `getSuccessOrThrow()`, which only accepts a function that throws an exception.

For an example:
```java
    for (final var entry : arguments.entrySet()) {
      try {
        switch (entry.getKey()) {
          case "x":
            template.x = this.mapper_x.deserializeValue(entry.getValue())
                .getSuccessOrThrow(failure -> new UnconstructableArgumentException("x", failure));
            break;
          case "y":
            template.y = this.mapper_y.deserializeValue(entry.getValue())
                .getSuccessOrThrow(failure -> new UnconstructableArgumentException("y", failure));
            break;
          case "vecs":
            template.vecs = this.mapper_vecs.deserializeValue(entry.getValue())
                .getSuccessOrThrow(failure -> new UnconstructableArgumentException("vecs", failure));
            break;
          default:
            invalidArgsExBuilder.withExtraneousArgument(entry.getKey());
        }
      } catch (final UnconstructableArgumentException e) {
        invalidArgsExBuilder.withUnconstructableArgument(e.parameterName, e.failure);
      }
    }
```

Using the `getSuccessOrThrow()` defined here: https://github.com/NASA-AMMOS/aerie/blob/develop/merlin-framework/src/main/java/gov/nasa/jpl/aerie/merlin/framework/Result.java?plain=1#L112

The `throws Throws` requires us to throw an exception if we want to cache/save `failure`. We could maybe change/add to `Result` to have a method that allows us to not `throws Throws`. Maybe I'm missing some other simpler option, open to other ideas :)